### PR TITLE
AIOS-5001 Loading indicator rotation layout fix

### PR DESCRIPTION
**Pull request by apeda**

### CODE REVIEW

File: Classes/IDMZoomingScrollView.m [hunk 1]
```diff
@@ -51,22 +51,55 @@ - (id)initWithPhotoBrowser:(IDMPhotoBrowser *)browser {
...
         _progressView.tag = 101;
         _progressView.thicknessRatio = 0.1;
         _progressView.roundedCorners = NO;
         _progressView.trackTintColor    = browser.trackTintColor    ? self.photoBrowser.trackTintColor    : [UIColor colorWithWhite:0.2 alpha:1];
         _progressView.progressTintColor = browser.progressTintColor ? self.photoBrowser.progressTintColor : [UIColor colorWithWhite:1.0 alpha:1];
+        _progressView.translatesAutoresizingMaskIntoConstraints = NO;
+
         [self addSubview:_progressView];
-        
+        [self addConstraint:[NSLayoutConstraint
```
Comment (jeremiah-de) [inline Classes/IDMZoomingScrollView.m]: @apeda can this use layout anchors instead?

File: Classes/IDMZoomingScrollView.m [hunk 2]
```diff
@@ -51,22 +51,55 @@ - (id)initWithPhotoBrowser:(IDMPhotoBrowser *)browser {
...
+        CGFloat progressViewDimension = 35.0;
+
+        _progressView = [[DACircularProgressView alloc] initWithFrame:CGRectMake(0.0, 0.0, progressViewDimension, progressViewDimension)];
         [_progressView setProgress:0.0f];
         _progressView.tag = 101;
         _progressView.thicknessRatio = 0.1;
         _progressView.roundedCorners = NO;
         _progressView.trackTintColor    = browser.trackTintColor    ? self.photoBrowser.trackTintColor    : [UIColor colorWithWhite:0.2 alpha:1];
         _progressView.progressTintColor = browser.progressTintColor ? self.photoBrowser.progressTintColor : [UIColor colorWithWhite:1.0 alpha:1];
+        _progressView.translatesAutoresizingMaskIntoConstraints = NO;
```
Comment (jeremiah-de) [inline Classes/IDMZoomingScrollView.m:64]: autolayout ftw!

File: Classes/IDMZoomingScrollView.m [hunk 3]
```diff
@@ -51,22 +51,55 @@ - (id)initWithPhotoBrowser:(IDMPhotoBrowser *)browser {
...
+        [self addConstraint:[NSLayoutConstraint
+                             constraintWithItem:_progressView
+                             attribute:NSLayoutAttributeHeight
+                             relatedBy:NSLayoutRelationEqual
+                             toItem:nil
+                             attribute:NSLayoutAttributeNotAnAttribute
+                             multiplier:1.0
+                             constant:progressViewDimension]];
+
+        [self layoutIfNeeded];
```
Comment (jeremiah-de) [inline Classes/IDMZoomingScrollView.m]: I think it's better to call setNeedsLayout instead of forcing a layout on init

**Comment by apeda** [comment]
@jeremiah-de Thanks for good advices! Please have a look at updated commit.

**Review by jeremiah-de — APPROVED** [review]

**Comment by ralph-e-boy** [comment]
@apeda  might be a good idea to make a PR of this fix against the real idm photo browser repo also, what do you think?

**Comment by apeda** [comment]
@ralph-e-boy It's a good idea to support open-source projects. But the fix we have uses UILayoutGuide which is available for iOS 9+, and original project claims iOS 8+ compatibility. So it seems they would not have possibility to accept it.

**Comment by ralph-e-boy** [comment]
@apeda  good point. maybe in a year then :-)

**Pull request merged**